tests/test_middleware.py

```diff
@@ -25,6 +25,12 @@ def process_response(self, req, resp, resource, req_succeeded):
         self.req_succeeded = req_succeeded
 
```

Comment:
Although not directly related to the other changes, this had to be addressed to unblock the coverage gate.